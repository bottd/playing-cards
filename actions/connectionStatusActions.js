export const actionTypes = {
  connectToGame: 'CONNECT_TO_GAME',
  disconnectFromGame: 'DISCONNECT_FROM_GAME',
}

export const connectToGame = () => ({
  type: actionTypes.connectToGame,
});

export const disconnectFromGame = () => ({
  type: actionTypes.disconnectFromGame,
});
