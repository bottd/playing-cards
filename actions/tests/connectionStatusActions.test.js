import { actionTypes, conectToGame, disconnectFromGame } from '../connectionStatusActions';

describe('Connection status actions', () => {
  describe('action types', () => {
    it('Should match snapshot', () => {
      expect(actionTypes).toMatchSnapshot();
    });
  });

  describe('connectToGame', () => {
    const expected = { type: 'CONNECT_TO_GAME' };
    const result = connectToGame();
    expect(result).toEqual(expected);
  });

  describe('disconnectFromGame', () => {
    const expected = { type: 'DISCONNECT_FROM_GAME' };
    const result = disconnectFromGame();
    expect(result).toEqual(expected);
  });
});
