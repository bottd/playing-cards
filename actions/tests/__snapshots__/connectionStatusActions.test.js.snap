// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection status actions action types Should match snapshot 1`] = `
Object {
  "connectToGame": "CONNECT_TO_GAME",
  "disconnectFromGame": "DISCONNECT_FROM_GAME",
}
`;
